tools/toolchains/arm.py

```diff
@@ -56,6 +56,11 @@ def __init__(self, target, options=None, notify=None, macros=None, silent=False,
         else:
             cpu = target.core
 
+        if not TOOLCHAIN_PATHS['ARM']:
+            exe = find_executable('armcc')
```

Comment:
Does this do the right thing on Linux/OSX and Windows (where the executable is called `armcc.exe`) ?